mergin/client_pull.py

```diff
@@ -183,6 +185,8 @@ def download_project_is_running(job):
     """
     for future in job.futures:
         if future.done() and future.exception() is not None:
+            job.mp.log.error("Error while downloading project: " + str(future.exception()))
+            job.mp.log.info("--- download aborted")
```

Comment:
I am thinking that as the handling of failed futures is the same in _is_running() and in _finalize() methods, maybe it would be good to move it to a new function to avoid code duplication? So here the _is_running() body would become just this:
1. _abort_download_on_failed_future()
2. test for future.running()